library/src/test/java/com/datatorrent/lib/stream/MultipleStreamMergerTest.java

```diff
@@ -0,0 +1,145 @@
+    operatorsToAdd = new ArrayList<>();
+
+  }
+
+  @Test
+  public void mergeTwoStreams()
+  {
+    RandomWordGenerator randomWordGenerator = new RandomWordGenerator();
+    RandomWordGenerator randomWordGenerator2 = new RandomWordGenerator();
+
+    randomWordGenerator.setTuplesPerWindow(1);
+    randomWordGenerator2.setTuplesPerWindow(1);
+
+    MultipleStreamMerger<byte[]> merger = new MultipleStreamMerger<>();
+    merger.merge(randomWordGenerator.output)
+        .merge(randomWordGenerator2.output);
+
+    merger.constructMergeTree(streamsToAddToDag, operatorsToAdd);
+
+    assertEquals("Count of created streams", 2, streamsToAddToDag.size());
+    assertEquals("Count of created operators", 1, operatorsToAdd.size());
+
+    // Next check actual connections
+    assertEquals("Generator 1 stream", randomWordGenerator.output,
+        streamsToAddToDag.get(0).sourcePort);
+
+    assertEquals("Generator 2 stream", randomWordGenerator2.output,
+        streamsToAddToDag.get(1).sourcePort);
+
+    assertEquals("Final operator input_1", operatorsToAdd.get(0).merger.data1, streamsToAddToDag.get(0).destPort);
+    assertEquals("Final operator input_2", operatorsToAdd.get(0).merger.data2, streamsToAddToDag.get(1).destPort);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void mergeOneStream()
+  {
+    RandomWordGenerator randomWordGenerator = new RandomWordGenerator();
+    MultipleStreamMerger<byte[]> merger = new MultipleStreamMerger<>();
+    merger.merge(randomWordGenerator.output);
+    merger.constructMergeTree(streamsToAddToDag, operatorsToAdd);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void mergeZeroStream()
+  {
+    RandomWordGenerator randomWordGenerator = new RandomWordGenerator();
+    MultipleStreamMerger<byte[]> merger = new MultipleStreamMerger<>();
+    merger.constructMergeTree(streamsToAddToDag, operatorsToAdd);
+  }
+
+  static class Application implements StreamingApplication
+  {
+    @Override
+    public void populateDAG(DAG dag, Configuration conf)
+    {
+      LOG.debug("Application - PopulateDAG");
+      int count = 53;
+      RandomWordGenerator[] generators = new RandomWordGenerator[count];
+      MultipleStreamMerger<byte[]> merger = new MultipleStreamMerger<>();
+      for (int i = 0; i < count; i++) {
+        generators[i] = new RandomWordGenerator();
+        generators[i].setTuplesPerWindow(1);
+        dag.addOperator("Generator " + i, generators[i]);
+        merger.merge(generators[i].output);
+      }
+
+      merger.insertInto(dag, conf);
+
+      // This should connect all the relevant ports
+      dag.addModule("Merger", merger);
+
+      // And then we should see the output
+      ConsoleOutputOperator consoleOperator = dag.addOperator("console", new ConsoleOutputOperator());
+      dag.addStream("merger-console", merger.streamOutput, consoleOperator.input);
+    }
+  }
+
+  @Test
+  public void testApplication() throws IOException, Exception {
```

Comment:
Can you also add validation to this test? A very basic test could be done by controlling the number of objects sent by each RandomWordGenerator and counting them at the console operator.